cabal-install/Distribution/Client/Init.hs

```diff
@@ -150,16 +159,34 @@ maybeToFlag = maybe NoFlag Flag
 
 -- | Get the package name: use the package directory (supplied, or the current
 --   directory by default) as a guess.
```

Comment:
@cartazio  what do you think about

-- | Get the package name: use the package directory (supplied, or the current
--   directory by default) as a guess. It looks at the SourcePackageDb to avoid
--   using an existing package name.